### PR TITLE
Copy ResultProvider sources, rather than binaries

### PR DESCRIPTION
**Pull request by amcasey**
We need to produce two copies of each ResultProvider DLL, one that runs on
NetFX20 and one that is Portable.  We originally believed that we could
create a single DLL that would work for both and so we just copied the
output of each ResultProvider project and signed it with a different
Authenticode certificate.  Now we know that - even though the source of
both versions will be the same - we need to actually recompile for
Portable.

As a first step, we eliminate the copying mechanism and split each
ResultProvider project in two.  To minimize duplication, we introduce
shared projects that hold on to most of the source.

The next step will be to make the two copies reference different
assemblies.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @cston @KevinH-MS @ManishJayaswal

File: src/ExpressionEvaluator/CSharp/Source/ResultProvider/NetFX20/CSharpResultProvider.NetFX20.csproj [hunk 1]
```diff
@@ -4,39 +4,35 @@
...
+    <Import Project="..\..\..\..\..\Tools\Microsoft.CodeAnalysis.Toolset.Open\Targets\VSL.Settings.targets" />
+    <Import Project="..\..\..\..\..\..\build\VSL.Settings.Closed.targets" />
   </ImportGroup>
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
     <ProjectGuid>{BF9DAC1E-3A5E-4DC3-BB44-9A64E0D4E9D3}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <AuthenticodeCertificateName>MicrosoftSHA1Win8WinBlue</AuthenticodeCertificateName>
+    <AuthenticodeCertificateName>WindowsPhone623</AuthenticodeCertificateName>
```
Comment (amcasey) [inline src/ExpressionEvaluator/CSharp/Source/ResultProvider/NetFX20/CSharpResultProvider.NetFX20.csproj]: This should not have changed.

**Comment by KevinH-MS** [comment]
:+1:

**Comment by amcasey** [comment]
I believe the Jenkins failure was due to ildasm not being able to write to a non-existent directory.  I've added a MakeDir invocation to create the directory.

**Pull request merged**